tests/unit/test_redfish.py

```diff
@@ -2,19 +2,108 @@
 from time import sleep
 from unittest.mock import Mock, patch
 
-from redfish.rest.v1 import InvalidCredentialsError, SessionCreationError
+import redfish_utilities
+from redfish.rest.v1 import (
+    InvalidCredentialsError,
+    RetriesExhaustedError,
+    SessionCreationError,
+)
 
 from prometheus_hardware_exporter.collectors.redfish import RedfishHelper
 from prometheus_hardware_exporter.config import Config
 
 
-class TestRedfishSensors(unittest.TestCase):
-    """Test the RedfishSensors class."""
+class TestRedfishMetrics(unittest.TestCase):
+    """Test metrics methods in RedfishHelper."""
 
-    def mock_helper(self):
-        mock_config = Mock()
-        return RedfishHelper(mock_config)
+    @patch("prometheus_hardware_exporter.collectors.redfish.redfish.redfish_client")
+    def test_00_redfish_helper_context_manager_success(self, mock_redfish_client):
+        mock_redfish_login = Mock()
+        mock_redfish_logout = Mock()
+        mock_redfish_client.return_value.login = mock_redfish_login
+        mock_redfish_client.return_value.logout = mock_redfish_logout
+        mock_config = Config(
+            redfish_host="",
+            redfish_username="",
+            redfish_password="",
+            redfish_client_timeout=10,
+            redfish_client_max_retry=5,
+            redfish_discover_cache_ttl=5,
+        )
+        with RedfishHelper(mock_config):
+            mock_redfish_client.assert_called_once_with(
+                base_url="",
+                username="",
+                password="",
+                timeout=10,
+                max_retry=5,
+            )
+            mock_redfish_login.assert_called_once_with(auth="session")
+        mock_redfish_logout.assert_called_once()
+
+    @patch("prometheus_hardware_exporter.collectors.redfish.redfish.redfish_client")
+    def test_01_redfish_helper_context_manager_fail(self, mock_redfish_client):
```

Comment:
**nitpick (blocking):** 

The purpose of this test function maybe need to redefine?
We are not testing any error handling here.

We maybe want to test the error handling in collector?

Reply:
We are testing whether the right exception is being raised if there's an error while getting the redfish object at the start of the context manager execution:

```python
# test_redfish.py

for err in [
    InvalidCredentialsError(),
    ConnectionError(),
    SessionCreationError(),
    RetriesExhaustedError(),
]:
    mock_redfish_client.side_effect = err
    print(type(err))
    with self.assertRaises(
        (
            InvalidCredentialsError,
            ConnectionError,
            SessionCreationError,
            RetriesExhaustedError,
        )
    ):
        with RedfishHelper(mock_config):
            pass
```

This code loops through each of the possible exceptions and then opens the RedfishHelper context manager. We are asserting that each of these exception are being raised (from the `__enter__` method of RedfishHelper) using the outer context manager for `self.assertRaises`.

Reply:
You have this line`mock_redfish_client.side_effect = err`. Of course it will raise the error.

You should test the error handling try catch instead of this.

Reply:
The line `mock_redfish_client.side_effect = err` is mocking the exception only for the redfish_client. I'm testing whether that it is being raised during the usage of the context manager for RedfishHelper also. 

> You should test the error handling try catch instead of this.
I'm also testing that in test_collector.py but only for ConnectionError. I'll add the other exceptions as well there.